### PR TITLE
Narayana: support @DontRollback on exception type

### PR DESCRIPTION
**Pull request by FroMage**
I need to be able to specify exception types that don't cause a rollback, such as redirect exceptions, for Renarde. This allows me to do it.

### CODE REVIEW

File: extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/DontRollback.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+/**
+ * If you annotate your exception with this annotation, the transactional interceptor will not
+ * rollback when your exception it thrown.
+ * 
+ * FIXME: move to SPI so it can be used by TransactionalReactive
+ */
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+public @interface DontRollback {
```
Comment (stuartwdouglas) [inline extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/DontRollback.java:18]: I think this should also support forcing a rollback (@Rollback(true) and @Rollback(false)). 

At the moment rollback happens for unchecked exceptions, and it does not roll back for checked, so we should be able to override both these behaviors.

File: extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/DontRollback.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+/**
+ * If you annotate your exception with this annotation, the transactional interceptor will not
+ * rollback when your exception it thrown.
+ * 
+ * FIXME: move to SPI so it can be used by TransactionalReactive
+ */
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
```
Comment (stuartwdouglas) [inline extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/DontRollback.java:17]: Should this be @Inherited ?

**Review by stuartwdouglas — CHANGES_REQUESTED** [review]

File: extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/DontRollback.java (hunk 1 above)
Comment (FroMage) [reply extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/DontRollback.java]: OK done.

File: extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/DontRollback.java (hunk 2 above)
Comment (FroMage) [reply extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/DontRollback.java]: Done

**Comment by FroMage** [comment]
Done, and added a test.

**Comment by FroMage** [comment]
Failure seems unrelated, let's restart.

**Comment by FroMage** [comment]
Same failure, again looks unrelated.

**Review by stuartwdouglas — APPROVED** [review]

**Comment by josephearl** [comment]
How does this differ from `@Transactional` and the `dontRollbackOn` parameter (which seems more flexible since I can choose to not rollback on exception types that are not under my control)?

**Comment by stuartwdouglas** [comment]
It is applied to the exception type rather than the interception point. With @Transactional if you have 100 endpoints you need to specify `dontRollbackOn` on every one.

**Comment by FroMage** [comment]
Yeah. And now I'm realising that I need the same `@Rollback` annotation for `@ReactiveTransactional` so I wonder if I duplicate it or create a new module for common annotations or move it to a more core package. WDYT @stuartwdouglas ?

**Comment by stuartwdouglas** [comment]
I guess a new common module that can be shared.

**Pull request merged**